### PR TITLE
server: proxy: support server heartbeats

### PR DESCRIPTION
**Pull request by kubistika**
I noticed that `rdpRdp` struct contains a pointer to another *empty* struct called `rdpHeartbeat`. So, I just added a callback called `ServerHeartbeat` and called it whenever the client receives a server heartbeat. I also added server functionality to send an heartbeat. I'm not sure I like the way the `rdpHeartbeat` struct is currently used, because I don't think it's place is under `rdpRdp`, as only clients (freerdp* structs) really receive heartbeats (from servers). I'd maybe remove this struct from `rdpRdp` and maintain it only in `freerdp` struct. What do you guys think?

### CODE REVIEW

File: include/freerdp/heartbeat.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+{
+	pServerHeartbeat ServerHeartbeat;
+};
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+	FREERDP_API int freerdp_heartbeat_send_heartbeat_pdu(freerdp_peer* peer, BYTE period,
```
Comment (akallabeth) [inline include/freerdp/heartbeat.h:39]: I´d opt for `BOOL` return as the only return is `-1` and 0 anyway

File: libfreerdp/core/heartbeat.c [hunk 2]
```diff
@@ -44,6 +44,40 @@ int rdp_recv_heartbeat_packet(rdpRdp* rdp, wStream* s)
 	         "received Heartbeat PDU -> period=%" PRIu8 ", count1=%" PRIu8 ", count2=%" PRIu8 "",
 	         period, count1, count2);
 
+	if (rdp->heartbeat)
+	{
+		BOOL rc = rdp->heartbeat->ServerHeartbeat(rdp->instance, period, count1, count2);
```
Comment (akallabeth) [inline libfreerdp/core/heartbeat.c:49]: Screams for `IFCALL` 
I think most of the time the callback will be `NULL` crashing here? Except in proxy, of course.

Comment (kubistika) [reply libfreerdp/core/heartbeat.c:49]: You're right, thanks

File: include/freerdp/heartbeat.h (hunk 1 above)
Comment (kubistika) [reply include/freerdp/heartbeat.h:39]: K

**Pull request merged**